### PR TITLE
Create separate CI jobs

### PR DESCRIPTION
**Pull request by benceharomi**
* `build` and `test` have been split to two separate CI jobs
* Configs added/updated for `EditorConfig`, `Prettier` and `Solhint` to enforce standardised formatting and linting
* Linting script added to the `package.json`
* Created a `lint` CI job that runs in parallel with `build`
* Linted all Solidity files in the `ethereum` directory:
  * the linting of the Solidity files happened with an automated tool (prettier)
  * prettier is not formatting too long [comments](https://prettier.io/docs/en/rationale.html#comments) and [imports](https://prettier.io/docs/en/rationale.html#imports), these were wrapped/skipped manually
* Added `max-warnings` as an upper limit to restrict introducing new warnings

### CODE REVIEW

File: .github/workflows/license.yaml
Comment (yorik) [inline .github/workflows/license.yaml:1]: Please note that in main branch this file was changed and renamed into [nodejs-license.yaml](https://github.com/matter-labs/era-contracts/blob/main/.github/workflows/nodejs-license.yaml).

Also it's just a copy from https://github.com/matter-labs/gh-workflows/blob/main/templates/.github/workflows/nodejs-license.yaml

Comment (benceharomi) [reply .github/workflows/license.yaml:1]: To avoid future conflicts, I removed the commit renaming the License Check job

**Comment by mm-zk** [comment]
Are all the changes to .sol files done by prettier?

If yes - can you do just this part in a separate PR.

**Review by mm-zk — APPROVED** [review]
For future - please clearly mark in PR description, that all solidity changes were done with automated tool.

**Review by vladbochok — APPROVED** [review]

**Comment by benceharomi** [comment]
> For future - please clearly mark in PR description, that all solidity changes were done with automated tool.

Description modified

**Pull request merged**